src/Core/Model/Common/AssetDraft.php

```diff
@@ -37,4 +37,36 @@ public function fieldDefinitions()
             'custom' => [static::TYPE => CustomFieldObjectDraft::class],
         ];
     }
+
+    /**
+     * @param AssetSourceCollection $sources
+     * @param LocalizedString $name
+     * @param Context|callable $context
+     * @return AssetDraft
+     */
+    public static function ofSourcesAndName(AssetSourceCollection $sources, LocalizedString $name, $context = null)
```

Comment:
maybe better change the order of parameters to $name and $sources
```suggestion
    public static function ofNameAndSources(LocalizedString $name, AssetSourceCollection $sources, $context = null)
```